Fix compilation issues found with recent `gcc` and `CMake` versions

- [x] A clear description of the issue

ProxySQL fails to compile with the following two errros:

1\. mariadb-client-library

```
-- Looking for pthread_getspecific - not found
-- Looking for pthread_getspecific in pthread
-- Looking for pthread_getspecific in pthread - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - found
-- Looking for setsockopt
-- Looking for setsockopt - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found OpenSSL: /home/javjarfer/Projects/proxysql_git/deps/libssl/openssl/lib/libcrypto.so (found version "3.0.0")
-- TLS library/version: OpenSSL 3.0.0
-- SYSTEM_LIBS dl;m;pthread;pthread;dl;m;/home/javjarfer/Projects/proxysql_git/deps/libssl/openssl/lib/libssl.so;/home/javjarfer/Projects/proxysql_git/deps/libssl/openssl/lib/libcrypto.so
-- Dynamic column API support: ON
-- SYSTEM processor: x86_64
CMake Error at cmake/ConnectorName.cmake:30 (ENDMACRO):
  Flow control statements are not properly nested.
Call Stack (most recent call first):
  CMakeLists.txt:434 (INCLUDE)

-- Configuring incomplete, errors occurred!
```

2. prometheus-cpp

```
[ 57%] Building CXX object core/CMakeFiles/core.dir/src/histogram.cc.o
[ 63%] Building CXX object core/CMakeFiles/core.dir/src/registry.cc.o
[ 68%] Building CXX object core/CMakeFiles/core.dir/src/serializer.cc.o
[ 73%] Building CXX object core/CMakeFiles/core.dir/src/summary.cc.o
[ 78%] Building CXX object core/CMakeFiles/core.dir/src/text_serializer.cc.o
/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp/core/src/histogram.cc: In member function ‘prometheus::ClientMetric prometheus::Histogram::Collect() const’:
/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp/core/src/histogram.cc:54:38: error: ‘numeric_limits’ is not a member of ‘std’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                      ^~~~~~~~~~~~~~
/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp/core/src/histogram.cc:54:53: error: expected primary-expression before ‘double’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                                     ^~~~~~
/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp/core/src/histogram.cc:54:53: error: expected ‘:’ before ‘double’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                                     ^~~~~~
      |                                                     :
/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp/core/src/histogram.cc:54:53: error: expected primary-expression before ‘double’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                                     ^~~~~~
/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp/core/src/histogram.cc:54:53: error: expected ‘)’ before ‘double’
   54 |                               ? std::numeric_limits<double>::infinity()
      |                                                     ^~~~~~
      |                                                     )
/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp/core/src/histogram.cc:53:26: note: to match this ‘(’
   53 |     bucket.upper_bound = (i == bucket_boundaries_.size()
      |                          ^
make[4]: *** [core/CMakeFiles/core.dir/build.make:188: core/CMakeFiles/core.dir/src/histogram.cc.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: Leaving directory '/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp-0.9.0'
[ 78%] Built target civetweb
make[4]: Leaving directory '/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp-0.9.0'
make[3]: *** [CMakeFiles/Makefile2:118: core/CMakeFiles/core.dir/all] Error 2
make[3]: Leaving directory '/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp-0.9.0'
make[2]: *** [Makefile:136: all] Error 2
make[2]: Leaving directory '/home/javjarfer/Projects/proxysql_git/deps/prometheus-cpp/prometheus-cpp-0.9.0'
make[1]: *** [Makefile:229: prometheus-cpp/prometheus-cpp/lib/libprometheus-cpp-core.a] Error 2
make[1]: Leaving directory '/home/javjarfer/Projects/proxysql_git/deps'
make: *** [Makefile:88: build_deps] Error 2
make build_deps -j$(nproc)  12.47s user 1.17s system 406% cpu 3.358 total
```

- [x] ProxySQL version

v2.x

- [x] OS version

Archlinux

- [x] The steps to reproduce the issue

Try to compile ProxySQL.

--- a/deps/Makefile
+++ b/deps/Makefile
@@ -173,6 +173,7 @@ mariadb-client-library/mariadb_client/libmariadb/libmariadbclient.a: libssl/open
 	cd mariadb-client-library && rm -rf mariadb-connector-c-3.1.9-src
 	cd mariadb-client-library && tar -zxf mariadb-connector-c-3.1.9-src.tar.gz
 	cd mariadb-client-library/mariadb_client && patch ./plugins/auth/CMakeLists.txt < ../plugin_auth_CMakeLists.txt.patch
+	cd mariadb-client-library/mariadb_client && patch -p0 < ../ConnectorName.patch
 	cd mariadb-client-library/mariadb_client && cmake . -DOPENSSL_ROOT_DIR=$(shell pwd)/libssl/openssl/ -DOPENSSL_LIBRARIES=$(shell pwd)/libssl/openssl/ .
 #	cd mariadb-client-library/mariadb_client && cmake . -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl # this is needed on MacOSX
 	cd mariadb-client-library/mariadb_client && patch libmariadb/mariadb_stmt.c < ../mariadb_stmt.c.patch

--- a/deps/Makefile
+++ b/deps/Makefile
@@ -225,6 +225,7 @@ prometheus-cpp/prometheus-cpp/lib/libprometheus-cpp-core.a:
 	cd prometheus-cpp && tar --strip-components=1 -zxf civetweb-v1.11.tar.gz -C prometheus-cpp/3rdparty/civetweb
 	cd prometheus-cpp/prometheus-cpp && patch -p1 < ../serial_exposer.patch
 	cd prometheus-cpp/prometheus-cpp && patch -p0 < ../registry_counters_reset.patch
+	cd prometheus-cpp/prometheus-cpp && patch -p0 < ../include_limits.patch
 	cd prometheus-cpp/prometheus-cpp && cmake . -DBUILD_SHARED_LIBS=OFF -DENABLE_TESTING=OFF -DENABLE_PUSH=OFF
 	cd prometheus-cpp/prometheus-cpp && CC=${CC} CXX=${CXX} ${MAKE}
 

--- a/deps/mariadb-client-library/ConnectorName.patch
+++ b/deps/mariadb-client-library/ConnectorName.patch
@@ -1,0 +1,13 @@
+diff --git cmake/ConnectorName.cmake cmake/ConnectorName.cmake
+index b7bbbad..357b8ac 100644
+--- cmake/ConnectorName.cmake
++++ cmake/ConnectorName.cmake
+@@ -22,7 +22,7 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Windows")
+     SET(MACHINE_NAME "x64")
+   ELSE()
+     SET(MACHINE_NAME "32")
+-  END()
++  ENDIF()
+ ENDIF()
+ 
+ SET(product_name "mysql-connector-c-${CPACK_PACKAGE_VERSION}-${PLATFORM_NAME}${CONCAT_SIGN}${MACHINE_NAME}")

--- a/deps/prometheus-cpp/include_limits.patch
+++ b/deps/prometheus-cpp/include_limits.patch
@@ -1,0 +1,12 @@
+diff --git core/src/histogram.cc core/src/histogram.cc
+index 8023529..b298fd4 100644
+--- core/src/histogram.cc
++++ core/src/histogram.cc
+@@ -3,6 +3,7 @@
+ #include <algorithm>
+ #include <cassert>
+ #include <iterator>
++#include <limits>
+ #include <numeric>
+ #include <ostream>
+ 

